sycl/source/detail/image_impl.cpp

```diff
@@ -482,6 +482,33 @@ void image_impl::unsampledImageDestructorNotification(void *UserObj) {
   XPTIRegistry::unsampledImageDestructorNotification(UserObj);
 }
 
+void image_impl::verifyProps(const property_list &Props) const {
+  auto CheckDataLessProperties = [](int PropertyKind) {
+#define __SYCL_DATA_LESS_PROP(NS_QUALIFIER, PROP_NAME, ENUM_VAL)               \
```

Comment:
Likewise. Maybe even provide an overload that only takes one lambda as a parameter and then delegates to the two-functors overload.

Reply:
discussed in private, not possible to implement this as suggested. resolving